gcloud/orb.yaml

```diff
@@ -88,6 +88,53 @@ commands:
       - run: ln -s /root/google-cloud-sdk/bin/gsutil /usr/bin/gsutil
       - run: ln -s /root/google-cloud-sdk/bin/docker-credential-gcloud /usr/bin/docker-credential-gcloud
 
+  patch-configmap-entry:
+    description: >
+      Update (or add) an entry in a configmap as a raw blob. Preserves all
+      other entries.
+    parameters:
+      configmap:
+        description: >
+          Name of the target k8s configmap.
+        type: string
+      entry_data:
+        description: >
+          File that contains the entry data.
+        type: string
+      entry_name:
+        description: >
+          Name of the entry in the configmap which exists under the top-level
+          "data" entry.
+        type: string
+    steps:
+      - run:
+          name: construct configmap patch file
+          command: |
+            # Avoid escaping issues by using input data as raw (|) directly
+            # from file. Must indent raw config (using sed here).
+            echo -e "data:\n  <<parameters.entry_name>>: |\n$(sed 's/^\(.*\)/    \1/g' <<parameters.entry_data>>)" > patch.yaml
+      - run:
+          name: patch entry in configmap
+          command: |
+            kubectl patch configmap/<<parameters.configmap>> \
+              --type merge -p "$(cat patch.yaml)"
+
+  force-redeploy-gke:
+    description: >
+      Force a k8s deployment to redeploy, which causes all pods to be
+      recreated.
+    parameters:
+      deployment:
+        description: >
+          Name of the k8s deployment.
+        type: string
+    steps:
+      - run:
+          name: poke deployment to force a redeploy
+          command: |
+            kubectl patch deployment <<parameters.deployment>> \
+              -p '{"spec":{"template":{"metadata":{"annotations":{"date":"'$(date +'%s')'"}}}}}'
```

Comment:
What is this for?  Is this the "poke the autoscaley boi" code?  Is there another use case, because if we're going to be hot-reloading the autoscaler, this might be unnecessary 🤔

Reply:
That's right, for now this will be for giving the autoscaler a good ol' poke to pick up the configmap changes. Until I sort out the inotify changes to the autoscaler to do lightweight hot reloads, we can't rely on hot reloads existing. But you're right, once that exists and we enable it this command will be not-so-useful for us.